Reorganize sections and add doc for filters and d&d (#283)
exoplatform/exo-documentation: docs/ManageTasks.rst

- At least one participant, which could be a user or a group
- A description which is not mandatory

.. _SearchProject:

How to search for a project?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> You can search for a project throught the filter field using a keyword. Results will contain projects having the keyword in their titles or description.

Reviewer: ```suggestion
You can find a projects through the filter field by entering a keyword contained in the project title or description.
```

.. _AccessProjectTasks:

How to access a project's tasks?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To access the list of a project's tasks, you just need to click on it's title on the CARD. You can choose between the two views BOARD or LIST.

Reviewer: ```suggestion
To access the list of tasks in a project, just click the project title on the CARD. Two views are available for tasks : BOARD and LIST.
```

> By default, project's tasks are displayed on the BOARD view.

Reviewer: ```suggestion
By default, tasks are displayed in the BOARD.
```

|NewTask09|

.. _AddTask:

How to add a new task?
~~~~~~~~~~~~~~~~~~~~~~~~

> You can add a task either in your Tasks list i.e. under TASKS tab or under a project in which you are at least participant. 

Reviewer: ```suggestion
You can add a task either in your Tasks list (i.e. under TASKS tab) or under a project in which you are already a participant or manager. 
```

> For that you just need to click on "Add Task" button |NewTask07| on the top left of the page.

Reviewer: ```suggestion
Click on "Add Task" button |NewTask07| .
```

A drawer opens allowing you to fill in the tasks' fields. 

|NewTask08|

> .. tip:: The title field is mandatory to enable tasks's creation.

Reviewer: ```suggestion
.. tip:: Only the title field is mandatory save a task.
```

.. note:: The differences between a task added under TASKS tab and under a project:
          

>           -  In TASKS tab, only the assignee filed is pre-filled, and by default assigned to you.

Reviewer: ```suggestion
          -  In TASKS tab, the task is pre-assigned to you by default.
```

>           -  Under a project, only the project field is pre-filled, by default set to the project under which you are adding a new task.

Reviewer: ```suggestion
          -  Within a project, the project field pre-filled with this project.
```

You can edit a task by clicking on its title which opens the "Edit task" drawer.

|NewTask10|

.. _SearchTask:

How to search for a task?
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Searching for a task is made the same in both tabs: TASKS and under a project.

Reviewer: ```suggestion
Searching for a task works the same within the TASKS tab and within a project.
```

> To search for a task you can either use the filter field allowing to search a task through a keyword or use the filter drawer which allows to make 

Reviewer: ```suggestion
To quickly find a task, you can use the filter field, to filter based on the list of tasks displayed. But for more
```

> an advanced search with multiple criteria. It also allows to group and sort tasks.

Reviewer: ```suggestion
advanced cases, you can use the filter button to select multiple criteria as well as grouping and sorting options.
```

.. _SearchTaskFilterField:

> Search for a task using the Filter field

Reviewer: ```suggestion
Finding a task using the Filter field
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To use the first alternative .i.e. the filter field, you just need to type a keyword in the field. The returned results correspond to tasks containing the keyword in the 

Reviewer: ```suggestion
To use the first alternative .i.e. the filter field, you just need to type a keyword in the field. The returned results correspond to tasks containing the keyword in their 
```

> title or the description.

Reviewer: ```suggestion
title or description.
```

|NewTask06|

.. _SearchTaskFilterBtn:

> Search for a task using the Filter button

Reviewer: ```suggestion
Finding a task using the Filter button
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The filter button allows you to make an advanced search and/or sort by setting the desired choices in 3 differents tabs.

Reviewer: ```suggestion
The filter button opens a drawer with more advanced criteria, grouping and sorting options.
```

Group and Sort tab
````````````````````
The group and sort tab allows you to group and/or sort tasks according to the selected choices.

|NewTask15|

> The "Group by" options organizes the tasks and groups them by a common criteria: Those belonging to the same project together, 

Reviewer: ```suggestion
The "Group by" option groups the resulting tasks by a common criteria such as due date, assignee, etc.
```

> those having the same due date (overdue, incoming...) together, etc...

Reviewer: ```suggestion
It's very useful to review the project from a certain angle
```

> The "Sort by" options allows to sort tasks in an order according to a criteria. For example when sorting by priority, tasks with the highest priority appear first.

Reviewer: ```suggestion
The "Sort by" options allows to sort tasks in an order according to a criteria. For example when sorting by priority, tasks with the highest priority appear first within their group.
```

> .. note:: Note that some critera in the "Group and Sort" tab differs between project's tasks and TASKS. FOr example, grouping project's tasks by project does not have sens

Reviewer: ```suggestion
.. note:: Note that some criteria in the "Group and Sort" tab differ between project's tasks and TASKS  tab. For example, grouping the tasks of a project by project does not make sense, so that option is not offered.
```

>           as all the tasks belong to the same project.

Reviewer: ```suggestion

```

Filter tab
```````````

> The tab "Filter" allows you to make a custom search based on the precised criteria.

Reviewer: ```suggestion
The "Filter" lets you to run a custom search based on multiple criteria combined with a logical "AND".
```

|NewTask12|

> - Contains: When setting a keyword in that field, the filter will search for tasks containing that word in its title or description i.e. it does the same thing as 

Reviewer: ```suggestion
- Contains: Enter any keywords in that field to match thats that contain these words in their title or description. It works exactly like the
```

>   the filter field when it is the only filled criteria.

Reviewer: ```suggestion
  filter field when it is the only filled criteria.
```

> - Assignee: Allows to select a user and then to filter tasks assigned to him.

Reviewer: ```suggestion
- Assignee: Lets you filter tasks assigned to a specific user. Start typing a name to trigger a list of suggested assignees among the participants of this project.
```

> - Due: A dropdown menu allowing to filter tasks in accordance to their due date: Overdue, Today, Tomorrow, Upcoming. Default set to All to list all tasks.

Reviewer: ```suggestion
- Due: Let you filter tasks based on their due date. A dropdown menu offers shortcuts : All, Overdue, Today, Tomorrow, Upcoming. 
```

> - Priority: A dropdown menu allowing to filter tasks by a selected priority: High, Normal, Low, None. Default set to All to display the whole tasks list.

Reviewer: ```suggestion
- Priority: Lets you filter tasks for a given priority. A dropdown menu offers the options : All, High, Normal, Low, None. 
```

> - Show completed tasks: a radio button allowing to display or not the :ref:`finished tasks <finishtask>`, default set to No, i.e. completed tasks are not displayed.

Reviewer: ```suggestion
- Show completed tasks:  when turned on this option lets you display :ref:`completed tasks <finishtask>` as well as the others.
```

> In the drawer's footer, the buttons/link make the following:

Reviewer: ```suggestion
At the bottom of the drawer, three actions are available :
```

> - Confirm: Allows to confirm the selected/inputted data and proceed to the filtering

Reviewer: ```suggestion
- Confirm: Validates the criteria and applies the filters to the current board/list of tasks.
```

> - Cancel: Allows to clear the selected/inputted data and close the drawer

Reviewer: ```suggestion
- Cancel: Dismisses the drawer without applying the filters.
```

> - Reset link: Allows to clear all the selected/inputted data

Reviewer: ```suggestion
- Reset: Clears all criteria, resetting them to their default value.
```

Labels tab
````````````

> The "Labels" tab allows you to search for tasks labelled with the inputted one.

Reviewer: ```suggestion
The "Labels" tab allows you to filter tasks based on their labels.
```

|NewTask16|

> .. tip:: When inputting more than one label, search results will contain tasks labelled by all of the selected ones.

Reviewer: ```suggestion
.. tip:: Currently, labels are combined with a logical 'AND' meaning the filtered list/board will contain tasks that have all the selected labels.
```

.. _finishtask:

Mark a task as completed
~~~~~~~~~~~~~~~~~~~~~~~~~~

> When you are done on a task and that you want to mark it as completed, you need simply to click on the radio button near the task's title.

Reviewer: ```suggestion
When you are done on a task and that you want to mark it as completed, you need simply to click on the radio button near the task's title. The task will disappear from the list/board (unless you selected "Show completed tasks" option in the filter dropdown).
```

|NewTask13|

|NewTask14|

.. _ChangeStatus:

> How to change a task's status?

Reviewer: ```suggestion
How to change the status of a task?
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> When you want to change a task's from a column (status) to another, you can do it through two manners:

Reviewer: ```suggestion
To reflect the progress of a task, you can change its status. There are two ways of doing it:
```

> -  Edit the taks and change its status to the desired one

Reviewer: ```suggestion
-  Edit the task and change and select the corresponding status in their Status field.
```

> -  Drag the task from its first column and the drop it to the desired one. The drag&drop is feasible in both views BOARD and LIST.

Reviewer: ```suggestion
-  Drag the task and drop it to the desired status column (BOARD view) or grouping (LIST view) .
```
